### ISSUE TITLE
Tags are not prefixed with leading `v`

### ISSUE THREAD

**Issue by ialidzhikov** [istio/api#2178]
(NOTE: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities/>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>
)

**Bug description**
From [Go modules wiki in golang/go](https://github.com/golang/go/wiki/Modules):
> Modules must be semantically versioned according to semver, usually in the form v(major).(minor).(patch), such as v0.1.0, v1.2.3, or v1.5.0-rc.1. **The leading v is required.**

Currently the tags of `istio.io/api` are not prefixed with a leading `v` - https://github.com/istio/api/tags.
On the other hand the tags of `istio.io/client-go` are prefixed with a `v` - https://github.com/istio/client-go/tags.

When using the `istio.io/api` in a Go modules project, it cannot be properly translated to concrete version:

```go
module github.com/foo/bar

go 1.16

require (
	// ...
	istio.io/api v0.0.0-20210520012029-891c0c12abfd
	istio.io/client-go v1.10.1
	// ...
)
```

Every time a developer has to check what is actually the version that corresponds to this commit sha.

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[X] User Experience

**Expected behavior**
`istio.io/api` tags to be prefixed with leading `v`.

**Steps to reproduce the bug**
See above. Create a go modules project that imports `istio.io/api` and make sure that its version is written with sha.

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
Not applicable.

**How was Istio installed?**
Not applicable.

**Environment where bug was observed (cloud vendor, OS, etc)**
Not applicable.

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by ialidzhikov** [comment]
@howardjohn this issue is open since 2 years and there is no single reply from maintainer.
Currently we see that this issue breaks dependabot for our projects - see https://github.com/dependabot/dependabot-core/issues/6889.
Is it possible to prefix the tags with `v`?

From above:
> From [Go modules wiki in golang/go](https://github.com/golang/go/wiki/Modules):
> > Modules must be semantically versioned according to semver, usually in the form v(major).(minor).(patch), such as v0.1.0, v1.2.3, or v1.5.0-rc.1. **The leading v is required.**

**Comment by busser** [comment]
This is also an issue for projects that use bots to keep dependencies up to date. For instance, we use Renovate. Since we import this package, Renovate opens a PR after every commit made to this repository.

Adding a `v` prefix would allow us to use semver in our `go.mod`, which renovate would bump only after actual releases on this repo.

Is adding a `v` to future tags an issue for some reason?

**Comment by howardjohn** [comment]
Should be fixed in 1.19. Note that we will publish 2 tags, one with `v`. Also its only api and client-go; we don't want users importing istio/istio

**Comment by axel7born** [comment]
Hi @howardjohn, 
This seems not to fixed in general. 
There are no `v`-prefixed tags for 1.19.7 and 1.20.3
